### PR TITLE
Add docker images push by tag

### PR DESCRIPTION
**Pull request by ArthurKamalov**
This PR adds a new step, in the docker publishing job in the GitHub actions, that verifies the given tag of the repository and publishes appropriate image tags:
- custom image tag.
- or production tag (e.g. 2.1.25) with the `latest` tag. That's only the case when the repo tag matches a  pattern of the repo's production tag, e.g - `v2.1.25`

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.1.26
+2.1.25
```
Comment (Kami) [inline VERSION:1]: Was this accidental change?

File: .github/workflows/agent-build.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ on: [push]
 # Agent Docker image tests
 jobs:
   docker-json:
-    uses: scalyr/scalyr-agent-2/.github/workflows/docker-and-k8s.yml@master
+    uses: scalyr/scalyr-agent-2/.github/workflows/docker-and-k8s.yml@add-images-prod-push
```
Comment (Kami) [inline .github/workflows/agent-build.yml:9]: I assume this testing branch name change will be removed before merging, right? :)

File: VERSION (hunk 1 above)
Comment (ArthurKamalov) [reply VERSION:1]: I accidentally bumped the version in previous PR's and forgot to return it back :)

File: .github/workflows/agent-build.yml (hunk 2 above)
Comment (ArthurKamalov) [reply .github/workflows/agent-build.yml:9]: Right, it seems that's pointless and we need to use it from master)

File: VERSION (hunk 1 above)
Comment (Kami) [inline VERSION:1]: Why did you revert this change?

IIRC, we can't do internal build via Jenkins if the VERSION file is not set to this value.

**Review by Kami — APPROVED** [review]
Approving, but please make sure VERSION file stuff is sorted out / understood as discussed on Slack.

**Comment by Kami** [comment]
OK, all the builds are green :)
